core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java

```diff
@@ -95,7 +95,7 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
       DimensionRawColumnChunk[] rawColumnChunks, int rowNumber,
       int pageNumber, DataOutputStream dataOutputStream) throws IOException {
     byte[] currentVal = copyBlockDataChunk(rawColumnChunks, rowNumber, pageNumber);
-    if (!this.isDictionary) {
+    if (!this.isDictionary && !this.isDirectDictionary) {
```

Comment:
Please add test cases corresponding to the changes.

Reply:
okay